nlmod/mdims/mlayers.py

```diff
@@ -815,198 +775,158 @@ def fill_top_bot_kh_kv_at_mask(model_ds, fill_mask, gridtype="structured"):
             # top ligt onder bottom_filled -> laagdikte wordt 0
             # top ligt boven bottom_filled -> laagdikte o.b.v. bottom_filled
             mask_top = model_ds["top"] < bottom_filled
-            model_ds["bot"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["top"], bottom_filled
-            )
-            model_ds["kh"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["kh"][lay], kh_filled
-            )
-            model_ds["kv"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["kv"][lay], kv_filled
-            )
-
+            model_ds["botm"][lay] = xr.where(fill_mask * mask_top,
+                                             model_ds["top"],
+                                             bottom_filled)
         else:
             # top ligt onder bottom_filled -> laagdikte wordt 0
             # top ligt boven bottom_filled -> laagdikte o.b.v. bottom_filled
-            mask_top = model_ds["bot"][lay - 1] < bottom_filled
-            model_ds["bot"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["bot"][lay - 1], bottom_filled
-            )
-            model_ds["kh"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["kh"][lay], kh_filled
-            )
-            model_ds["kv"][lay] = xr.where(
-                fill_mask * mask_top, model_ds["kv"][lay], kv_filled
-            )
+            mask_top = model_ds["botm"][lay - 1] < bottom_filled
+            model_ds["botm"][lay] = xr.where(fill_mask * mask_top,
+                                             model_ds["botm"][lay - 1],
+                                             bottom_filled)
+        model_ds["kh"][lay] = xr.where(fill_mask * mask_top,
+                                       model_ds["kh"][lay],
+                                       kh_filled)
+        model_ds["kv"][lay] = xr.where(fill_mask * mask_top,
+                                       model_ds["kv"][lay],
+                                       kv_filled)
 
     return model_ds
 
 
-def update_model_ds_from_ml_layer_ds(
-    model_ds,
-    ml_layer_ds,
-    gridtype="structured",
-    keep_vars=None,
-    add_northsea=True,
-    anisotropy=10,
-    fill_value_kh=1.0,
-    fill_value_kv=0.1,
-    cachedir=None,
-):
-    """Update a model dataset with a model layer dataset.
+def complete_ds(ds, anisotropy=10.0, fill_value_kh=1.0, fill_value_kv=0.1,
+                remove_nan_layers=True):
```

Comment:
Function name is a bit vague. Don't have a better idea quite yet though...

`fill_nan_top_botm_kh_kv`?

Reply:
I agree, will change this